src/test/java/io/crate/consumer/CrateConsumer.java

```diff
@@ -0,0 +1,149 @@
+        client.sql(CrateConsumerConstants.CREATE_TABLE_IF_NOT_EXISTS_STMT).actionGet();
+    }
+
+    private static String[] getHosts() {
+        final String hostsStr = System.getProperty(CrateConsumerConstants.CRATE_HOST);
+        if (hostsStr != null && !hostsStr.trim().equals("")) {
+            return hostsStr.split(",");
+        }
+        throw new IllegalArgumentException(String
+                .format("Missing global property: %s", CrateConsumerConstants.CRATE_HOST));
+    }
+
+    private int getPort(String portProperty) {
+        final String port = System.getProperty(portProperty);
+        if (port != null && !port.trim().equals("")) {
+            return Integer.valueOf(port);
+        }
+        throw new IllegalArgumentException(String
+                .format("Missing global property: %s", portProperty));
+    }
+
+    private Stream<String> composeUrls(String[] hosts, int port) {
+        return Arrays.stream(hosts)
+                .map(host -> String.format("http://%s:%d", host, port));
+    }
+
+    @Override
+    public void accept(Result result) throws IOException {
+        try {
+            Object[] args = new Object[]{
+                    result.getShortTestClassName(),
+                    result.getTestMethodName(),
+                    crateBuildVersion(),
+                    crateBuildTimestamp(),
+                    benchmarkValues(result)
+            };
+            client.sql(new SQLRequest(CrateConsumerConstants.INSERT_STMT, args)).actionGet();
+        } catch (ParseException e) {
+            LOGGER.warn("Cannot parse timestamp: ", e);
+        } catch (IOException e) {
+            LOGGER.warn("Error while fetching crate build timestamp: ", e);
+        } catch (Exception e) {
+            LOGGER.warn("Result of benchmark run is not inserted: ", e);
+        }
+    }
+
+    private Map<String, Object> benchmarkValues(Result result) {
+        return new HashMap<String, Object>() {{
+            put("benchmark_rounds", result.benchmarkRounds);
+            put("warmup_rounds", result.warmupRounds);
+            put("round_avg", result.roundAverage.avg);
+            put("round_stddev", result.roundAverage.stddev);
+            put("gc_avg", result.gcAverage.avg);
+            put("gc_stddev", result.gcAverage.stddev);
+            put("gc_invocations", result.gcInfo.accumulatedInvocations());
+            put("gc_time", result.gcInfo.accumulatedTime());
+            put("benchmark_time_total", result.benchmarkTime);
+            put("warmup_time_total", result.warmupTime);
+        }};
+    }
+
+    private String crateBuildVersion() {
+        SQLResponse response = client.sql(CrateConsumerConstants.SELECT_VERSIONS).actionGet();
+        Object[][] rows = response.rows();
+        assert rows.length != 0;
+        checkClusterVersionsHomogeneity(rows);
+        return String.valueOf(rows[0][0]);
+    }
+
+    private void checkClusterVersionsHomogeneity(Object[][] rows) {
```

Comment:
Why is it in the responsibility of the consumer to check if the cluster is setup correctly?